src/compiler/declarationEmitter.ts

```diff
@@ -396,6 +398,12 @@ namespace ts {
                 }
             }
 
+            function emitTypePredicate(type: TypePredicateNode) {
+                writeEntityName(type.parameterName);

```

Comment:
That said, a parameter name cannot be a qualified name, so I would just use `writeTextOfNode` here instead of adding a new `writeEntityName`.